ocs_ci/utility/framework/base_initializer.py

```
@@ -0,0 +1,201 @@
+        """
+        logger.info("Testing access to cluster with %s", kubeconfig_path)
+        if not os.path.isfile(kubeconfig_path):
+            raise ClusterNotAccessibleError(
+                "The kubeconfig file %s doesn't exist!", kubeconfig_path
+            )
+        if not which("oc"):
+            get_openshift_client()
+        try:
+            exec_cmd(f"oc --kubeconfig {kubeconfig_path} cluster-info")
+        except CommandFailed as ex:
+            raise ClusterNotAccessibleError("Cluster is not ready to use: %s", ex)
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)